tasks/12-ibm-codeengine-deploy.yaml

```diff
@@ -0,0 +1,150 @@
+apiVersion: tekton.dev/v1alpha1
+kind: Task
+metadata:
+  name: ibm-codeengine-deploy
+spec:
+  params:
+    - name: image-to
+      type: string
+    - name: tools-image
+      default: garagecatalyst/ibmcloud-dev:1.1.3
+    - name: source-dir
+      default: /source
+  volumes:
+    - name: source
+      emptyDir: {}
+  stepTemplate:
+    volumeMounts:
+      - name: source
+        mountPath: $(params.source-dir)
+  steps:
+    - name: codeengine-deploy
+      image: $(params.tools-image)
+      workingdir: $(params.source-dir)
+      env:
+        - name: NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: APIKEY
+          valueFrom:
+            secretKeyRef:
+              name: ibmcloud-apikey
+              key: APIKEY
+              optional: true
+        - name: CE_DEPLOY
+          valueFrom:
+            secretKeyRef:
+              name: codeengine-access
```

Comment:
This tasks expects a secret named `codeengine-access` to exist with a particular structure. If the secret exists but it is missing some of the fields it appears this task will fail (e.g. if CE_DEPLOY is set to `true` but CE_PROJECT_NAME is missing then `ibmcloud ce project select -n ${CE_PROJECT_NAME}` will fail).

Please provide a file named 12-ibm-codeengine-deploy.md in the tasks/ folder with information about the structure of the secret and instructions on how to create it. If necessary, you can add a shell script in the bin/ folder to help with the process. (I know this is something new but we will be doing the same for the other tasks as well.)